### ISSUE TITLE
Geotiff Module

### ISSUE THREAD

**Issue by tech4GT** [publiclab/image-sequencer#962]
@jywarren If you happen to have any resources regarding this, it would be really helpful. :smile:

**Comment by jywarren** [comment]
Yes!

https://github.com/publiclab/mapknitter/issues/296

🙌🏽

We may need to fetch the coordinates from an earlier step though, so would
need to pass along metadata somehow? Or I guess we could specify it in the
initial multi image steps Json.

**Comment by tech4GT** [comment]
@jywarren So the idea of a module which is non chain-able is a little weird inside of Image sequencer. I was thinking maybe this can output the same input image as far as Image-sequencer is concerned but it maybe outputs the other data as a file and saves it. What do you think?

**Comment by harshkhandeparkar** [comment]
Why is it non chainable? Maybe we can return a new sequencer.

**Comment by jywarren** [comment]
Yeah I agree, but so the idea of an output module would be one which ends
the sequence. It could still output a data url, but just not one in a
format that another standard module could read. Later this idea could
encompass an export to svg or PDF module, for example.

**Comment by tech4GT** [comment]
@jywarren hmm, I’m not sure I get you completely. So I was thinking that this module can just output the input, sequencer wise but would start a file download in the browser or on the node side save a file. Is this what you are thinking too?

**Comment by jywarren** [comment]
I think that'll work fine, sure! In any case we will be wanting to send it
to Google cloud storage so that sounds good. If we ever wanted a data-url
we could do that separately.

I'm mainly suggesting a new standard mental model for the general behavior
of "doing things with what image-sequencer produces", with the idea of a
new type of module called an "output module", however we don't have to
address this right now. Thanks!

**Comment by tech4GT** [comment]
Aha got it! So this output module supposedly ends the sequence if applied right?

**Comment by jywarren** [comment]
Yeahhh, it's not super important but with more output module types emerging
I thought it might be a good idea to create some norms around and guidance.

**Comment by tech4GT** [comment]
@jywarren I suppose this should be our next step??

**Comment by tech4GT** [comment]
I think this is all that is left as far as Image Sequencer is concerned, correct?
Apart from this we have to make a converter function for the json request and implement the workflow but thats on the app, right?
@jywarren

**Comment by harshkhandeparkar** [comment]
Hey @tech4GT I had a question, puppeteer runs a headless version of chromium right?

**Comment by tech4GT** [comment]
Depends, `puppeteer-core` can utilize whatever browser you have installed, it does not explicitly download chromium.

**Comment by harshkhandeparkar** [comment]
Can it be forced or add a rule to prioritize using firefox? Can it even use firefox btw?

**Comment by tech4GT** [comment]
@jywarren would love your input here, thanks! :smile:

**Comment by jywarren** [comment]
Ah, well we do need the webgl-distort module in your new gl code, so we can
go through the full distortion + compositing sequence with each image,
right? Thanks!

**Comment by tech4GT** [comment]
Oh sure! Do you want me to make webgl-distort into a is-module. Actually My exams start next week so I want to get this done before that. I think we have most of the pieces, major work remains in glueing them together on the is-app, correct?

**Comment by jywarren** [comment]
Yes, that's right. I do think we need the webgl-distort module in order to
do this... I think we could take on the remainder while you're on exams,
though!

I think the Geotiff lib is less important -- it's a final export step once
the rest is running. Can we get the webgl-distort working, which I think is
the last major piece? Thank you!!!

**Comment by tech4GT** [comment]
Working on webgl-distort now!

**Comment by harshkhandeparkar** [comment]
@jywarren @tech4GT the geotiff module just exports the image as a `.tiff` right?

**Comment by jywarren** [comment]
I'm going to move this to image-sequencer-app!
